src/manifest.c

```diff
@@ -525,12 +551,30 @@ static int retrieve_manifests(int current, int version, char *component, struct
 	free(dir);
 
 	/* FILE is not set for a MoM, only for bundle manifests */
-	if (file && current < version) {
+	if (file && current < version && strcmp(component, "full") != 0) {
 		if (try_delta_manifest_download(current, version, component, file) == 0) {
 			return 0;
 		}
 	}
 
+	/* If it's mix content just hardlink instead of curl download */
+	if (is_mix) {
+		string_or_die(&filename, "%s/%i/Manifest.%s.tar", state_dir, version, component);
+		string_or_die(&url, "%s/%i/Manifest.%s.tar", basedir, version, component);
+		ret = link(url, filename);
+		/* Try doing a regular rename if hardlink fails */
```

Comment:
This link, if it fails call rename, ought to be its own function. It probably ought to have a "make a copy" as another fallback.

Reply:
I agree, it would be nice to this logic wrapped in a separate function. Should there be a copy fallback, @tmarcu ?

Reply:
Any comments from @tmarcu or @matthewrsj here? Do we need a copy fallback for this case?

Reply:
See comment (way) below, I don't think this feature needs a copy fallback in the current state that it's in.